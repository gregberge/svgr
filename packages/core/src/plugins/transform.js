import reactDomTemplate from '../templates/reactDomTemplate'
import reactNativeTemplate from '../templates/reactNativeTemplate'
import reactTypescriptTemplate from '../templates/reactTypescriptTemplate'

export default (code, config, state) => {
  let transform = reactDomTemplate
  if (config.native) transform = reactNativeTemplate
  if (config.typescript) transform = reactTypescriptTemplate
  if (config.template) transform = config.template
  return transform(code, config, state)
}
