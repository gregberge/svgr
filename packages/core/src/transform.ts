import { expandState } from './state'
import { loadConfig } from './config'
import { resolvePlugin, getPlugins } from './plugins'
import type { Config } from './config'
import type { State } from './state'

const run = (code: string, config: Config, state: Partial<State>): string => {
  const expandedState = expandState(state)
  const plugins = getPlugins(config, state).map(resolvePlugin)
  let nextCode = String(code).replace('\0', '')
  // eslint-disable-next-line no-restricted-syntax
  for (const plugin of plugins) {
    nextCode = plugin(nextCode, config, expandedState)
  }
  return nextCode
}

export const transform = async (
  code: string,
  config: Config = {},
  state: Partial<State> = {},
): Promise<string> => {
  config = await loadConfig(config, state)
  return run(code, config, state)
}

transform.sync = (
  code: string,
  config: Config = {},
  state: Partial<State> = {},
): string => {
  config = loadConfig.sync(config, state)
  return run(code, config, state)
}
