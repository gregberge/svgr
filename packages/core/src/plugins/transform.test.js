import transform from './transform'

describe('transform', () => {
  it('should wrap it into component', () => {
    const result = transform(
      `<div />`,
      { template: (code, config, state) => `${state.foo}${code}` },
      { foo: 'bar' },
    )

    expect(result).toBe('bar<div />')
  })
})
