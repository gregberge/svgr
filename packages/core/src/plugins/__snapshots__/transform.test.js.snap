// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should transform into typescript 1`] = `
"import * as React from 'react'

const Svg = () => <div />

export default Svg"
`;
